metrics/api/src/main/java/io/helidon/metrics/api/MeterRegistry.java

```diff
@@ -0,0 +1,197 @@
+     * @return meters which match the predicate
+     */
+    Collection<Meter> meters(Predicate<Meter> filter);
+
+    /**
+     * Returns previously-registered meters which match one of the specified scopes.
+     *
+     * @param scopeSelection scopes to match
+     * @return matching meters
+     */
+    Iterable<Meter> meters(Iterable<String> scopeSelection);
+
+    /**
+     * Returns the scopes (if any) as represented by tags on meter IDs.
+     *
+     * @return scopes across all registered meters
+     */
+    Iterable<String> scopes();
+
+    /**
+     * Returns whether the specified meter is enabled or not, based on whether the meter registry as a whole is enabled and also
+     * whether the config settings for filtering include and exclude indicate the specific meter is enabled.
+     *
+     * @param name name of the meter to check
+     * @param tags tags of the meter to check
+     * @param scope scope, if present, of the meter to check
+     * @return true if the meter (and its meter registry) are enabled; false otherwise
+     */
+    boolean isMeterEnabled(String name, Iterable<Tag> tags, Optional<String> scope);
+
+    /**
+     * Returns the default {@link io.helidon.metrics.api.Clock} in use by the registry.
+     *
+     * @return default clock
+     */
+    Clock clock();
+
+    /**
+     * Locates a previously-registered meter using the name and tags in the provided builder or, if not found, registers a new
+     * one using the provided builder.
+     *
+     * @param builder builder to use in finding or creating a meter
+     * @return the previously-registered meter with the same name and tags or, if none, the newly-registered one
+     * @param <B> builder for the meter
+     * @param <M> type of the meter
+     */
+    <B extends Meter.Builder<B, M>, M extends Meter> M getOrCreate(B builder);
+
+    /**
+     * Locates a previously-registered counter.
+     *
+     * @param name name to match
+     * @param tags tags to match
+     * @return {@link java.util.Optional} of the previously-registered counter; empty if not found
+     */
+    default Optional<Counter> getCounter(String name, Iterable<Tag> tags) {
+        return get(Counter.class, name, tags);
+    }
+
+    /**
+     * Locates a previously-registered distribution summary.
+     *
+     * @param name name to match
+     * @param tags tags to match
+     * @return {@link java.util.Optional} of the previously-registered distribution summary; empty if not found
+     */
+    default Optional<DistributionSummary> getSummary(String name, Iterable<Tag> tags) {
+        return get(DistributionSummary.class, name, tags);
+    }
+
+    /**
+     * Locates a previously-registered gauge.
+     *
+     * @param name name to match
+     * @param tags tags to match
+     * @return {@link java.util.Optional} of the previously-registered gauge; empty if not found
+     */
+    default Optional<Gauge> getGauge(String name, Iterable<Tag> tags) {
```

Comment:
Can we use just `gauge` as the method name? And instead of `get` just `metric`? We may have discussed this before - in that case disregard the change, but maybe add a line comment explaining why we use this naming pattern)

Reply:
Good catch. The `get*` methods were left over; earlier we had

```
Gauge gauge(name, tags) // get-or-create
Optional<Gauge> getGauge(name, tags) // get only
```

But now that I have changed the interface to use the single method `getOrCreate(builder)` in place of the type-specific get-or-create methods, the "get-only"methods can use our normal naming convention.

Reply:
changed